### PR TITLE
Remove stale bricks after snapshot restore

### PR DESCRIPTION
**Pull request by nnDarshan**
tendrl-bug-id: Tendrl/gluster-integration#518
Signed-off-by: Darshan N <darshan.n.2024@gmail.com>

### CODE REVIEW

**Comment by nnDarshan** [comment]
**NOT VERIFIED, PLS DONT MERGE**

File: tendrl/gluster_integration/message/callback.py [hunk 1]
```diff
@@ -676,6 +677,86 @@ def brick_replace(self, event):
...
+        # get the list of current bricks by running get-state
+        subprocess.call(
+            [
+                'gluster',
+                'get-state',
+                'glusterd',
+                'odir',
+                '/var/run',
+                'file',
+                'glusterd-state-snapshot',
```
Comment (r0h4n) [inline tendrl/gluster_integration/message/callback.py]: this file name needs to be unique per event, please add a uuid of something

File: tendrl/gluster_integration/message/callback.py [hunk 2]
```diff
@@ -676,6 +677,86 @@ def brick_replace(self, event):
...
+                    index, b_index
+                )
+                brick = raw_data['Volumes'][curr_brick]
+            except KeyError:
+                break
+            latest_bricks.append(brick)
+
+        # get the list of bricks in etcd for this volume
+
+        sub_volumes = NS._int.client.read(
```
Comment (r0h4n) [inline tendrl/gluster_integration/message/callback.py]: use commons etcd_utils.read

File: tendrl/gluster_integration/message/callback.py [hunk 3]
```diff
@@ -676,6 +677,86 @@ def brick_replace(self, event):
...
+                fqdn = brick.key.split('/')[-1].split(':')[0]
+                path = brick.key.split('/')[-1].split(':')[-1][1:]
+
+                brick_path = "clusters/{0}/Bricks/"\
+                             "all/{1}/{2}".format(
+                                 NS.tendrl_context.integration_id,
+                                 fqdn,
+                                 path
+                             )
+                brick_full_path = NS._int.wclient.read(
```
Comment (r0h4n) [inline tendrl/gluster_integration/message/callback.py]: Use commons etcd_utils.read

File: tendrl/gluster_integration/message/callback.py (hunk 1 above)
Comment (nnDarshan) [reply tendrl/gluster_integration/message/callback.py]: ack

File: tendrl/gluster_integration/message/callback.py (hunk 2 above)
Comment (nnDarshan) [reply tendrl/gluster_integration/message/callback.py]: ack

File: tendrl/gluster_integration/message/callback.py (hunk 3 above)
Comment (nnDarshan) [reply tendrl/gluster_integration/message/callback.py]: ack

**Comment by nnDarshan** [comment]
> Verified?

Yes, Verified

**Pull request merged**